File: Daily Coding Practice/2020-01-09/748 Letter Case Permutation.py
class Solution:
    def letterCasePermutation(self, S):
        res = []
        path = ''
        self.dfs(S, 0, res, path)
        return res
    
    def dfs(self, S, index, res, path):
        res.append(path)
        return
    
        if S[index].isalpah():
            self.dfs(S, index+1, res, path+S[index].upper())
            self.dfs(S, index+1, res, path+S[index].lower())
        else:
            self.dfs(S, index+1, res, path+S[index])

File: Daily Coding Practice/2019-10-30/748 Letter Case Permutation.cpp
#include<string>
#include<vector>
using namespace std;

class Solution {
public:
    vector<string> ans;
    vector<string> letterCasePermutation(string S) {
        dfs(S, 0);
        return ans;
    };
    
    void dfs(string S, int u)     
        {        
        if (u == S.size())
        {
            ans.push_back(S);
            return;
        }
        dfs(S, u+1);
        
        if (S[u] >= 'A')
        {
            S[u] ^= 32;
            dfs(S, u +1);
        }
    }
};